src/pybind/mgr/restful/module.py

```diff
@@ -495,7 +495,11 @@ def get_mons(self):
         # Add more information
         for mon in mon_map_mons:
             mon['in_quorum'] = mon['rank'] in mon_status['quorum']
-            mon['server'] = self.get_metadata("mon", mon['name'])['hostname']
+            metadata = self.get_metadata("mon", mon['name'])
+            if metadata is None:
+                mon['server'] = '[UNKNOWN]'
```

Comment:
Hm. `[UNKNOWN]` is kind of an unexpected and undocumented magic value. I'd expect that no client will handle this value properly, as there is a high probability that `[UNKNOWN]` never be returned in a testing environment. 

IMO, we should return something that is either clearly an error, or make the result syntactically valid e.g. by removing this server.

Reply:
@sebastian-philipp what do you mean by "clearly an error"? thought `[UNKNOWN]` is exactly what i'd image for an error.

Reply:
Well, the result could look like so:

```json
[
  {
    "in_quoum": true,
    "leader": false,
    "server": {...},
    ...
  },
  {
    "in_quoum": true,
    "leader": false,
    "server": {...},
    ...
  },
  {
    "in_quoum": true,
    "leader": false,
    "server": "[UNKNOWN]",
    ...
  }
]
```

This has multiple problems:


1. As a typical client, I might implement something like this
    ```python

    all_versions = [mon['server']['version'] for mon in mons]
    ```
    In this case this will raise a TypeError, which really looks like a bug in the client
    ```
    TypeError: string indices must be integers, not str
    ```
2. In order to detect, if the reply is complete and consistent, you have to look at each and every mon and compare `mons["server"]` to `"[UNKNOWN]"`. 
3. It also might generate some wired results. Lets say, you want to print the mon's metadata as a csv table. Let's generate the csv header:
    ```python

    all_columns = set(sum([set(mon['server']) for mon in mons]))

    ```
    In this case this will not generate any error, but silently generate garbage.

Reply:
I suppose we could return `None` or an empty string if we can't detect the server instead and let the client handle it and assign a value that it desires?